### DISCUSSION TITLE
Size reduction for embedded system

### DISCUSSION THREAD

**Discussion by andreca** [CESNET/libyang#1973] [General]
Hello. We are working on an embedded product that uses `frr` which in turn depends on `libyang`. Our system is size constrained and we are trying do reduce `frr`'s footprint. Do you have any suggestions/thoughts on what could be removed/opted out from `libyang` to reduce its footprint? Any low hanging fruit?

**Comment by michalvasko** [comment]
Please use the latest released version `2.1.30`. Version `2.1.4` was unintentionally large, which was fixed in the current release.

**Reply by andreca** [reply]
We are using `2.0.7` at the moment (meta-openembedded:honister) and get a size-optimized binary, about `600 KiB` in size.

**Reply by michalvasko** [reply]
Oh, I see. In that case I do not think I can help you. If 600 KB (which is an achievement in itself I think) is too much, you have what I would consider an extreme use-case.

**Reply by andreca** [reply]
I see, thank you for your prompt response. If we go down the path of *trying* to reduce the size, would it be of any interest to the project (as a PR)?

**Reply by michalvasko** [reply]
Depends on the actual changes, as long as they do not make much mess of anything then why not.
